### PR TITLE
v1.28.0 — Menu layout updates.

### PR DESCRIPTION
**Pull request by DamianMullins**
Waiting on [this Fozzie PR](https://github.com/justeat/fozzie/pull/152).

### Added
- Scroll spy module.

### Changed
- Updated menu layouts.

### CODE REVIEW

File: docs/src/templates/includes/components/core/menu/status/je-menu-status-delivery-open.hbs [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<p class="c-badge c-badge--info c-badge--rounded l-centered">
+<span class="c-badge c-badge--info c-badge--rounded c-badge--alignCenter">
```
Comment (ashleynolan) [inline docs/src/templates/includes/components/core/menu/status/je-menu-status-delivery-open.hbs]: What's the reason to change these to `span`'s?

Not sure if there are any accessibility implications for just using spans to wrap text (so no semantic tag wrapped around it)…?

Comment (DamianMullins) [reply docs/src/templates/includes/components/core/menu/status/je-menu-status-delivery-open.hbs]: It is wrapped in a `p` tag in the template which pulls this one in, unfortunately it's hard to tell by way the partials are separated out.

screen shot 2018-09-14 at 09 14 15

Comment (DamianMullins) [reply docs/src/templates/includes/components/core/menu/status/je-menu-status-delivery-open.hbs]: https://github.com/justeat/global-component-library/pull/58/files#diff-43ffbb6c9652e23638fa60577b748b76R44

File: assets/src/js/ScrollSpy.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                if (progress >= this.duration) progress = this.duration;
+                if (progressPercentage >= 1) progressPercentage = 1;
+                const perTick = startingY + (easing(progressPercentage) * (difference - this.offset));
+
+                window.scrollTo(0, perTick);
+
+                if (progress < this.duration) {
+                    this.scrollAnimationFrame = window.requestAnimationFrame(step);
+                } else {
+                    window.addEventListener('scroll', this.onScrollHandler);
```
Comment (sonnyprince) [inline assets/src/js/ScrollSpy.js]: Does this need throttling?

File: assets/src/js/ScrollSpy.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        let currItem;
+
+        this.items.forEach(item => {
+            const hash = ScrollSpy.getHash(item);
+            const target = document.getElementById(hash.substr(1));
+
+            if (!target) return;
+
+            const isScreenPastSection =
+                window.pageYOffset >= ScrollSpy.getOffsetTop(target) - this.offset;
```
Comment (sonnyprince) [inline assets/src/js/ScrollSpy.js]: is it worth setting the `window.pageYOffset` as a const outside of the foreach?

File: assets/src/js/ScrollSpy.js (hunk 2 above)
Comment (DamianMullins) [reply assets/src/js/ScrollSpy.js]: Done, I applied the throttling up in the constructor. Has to be there so that the event can be removed correctly 😄

**Pull request merged**